bundles/CoreBundle/src/DependencyInjection/Compiler/ContainerAwarePass.php

```diff
@@ -0,0 +1,74 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * This source file is available under the terms of the
+ * Pimcore Open Core License (POCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (https://www.pimcore.com)
+ *  @license    Pimcore Open Core License (POCL)
+ */
+
+namespace Pimcore\Bundle\CoreBundle\DependencyInjection\Compiler;
+
+use ReflectionClass;
+use Symfony\Component\DependencyInjection\Reference;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Pimcore\Bundle\CoreBundle\DependencyInjection\ContainerAwareInterface;
+
+/**
+ * Automatically injects the service container into all services implementing ContainerAwareInterface.
+ *
+ * This compiler pass provides backward compatibility for services that need access to the
+ * service container, similar to Symfony 6's behavior with ContainerAwareInterface.
+ *
+ * Note: This is deprecated by design. New code should use dependency injection instead.
+ *
+ * @internal
+ */
+final class ContainerAwarePass implements CompilerPassInterface
+{
+    public function process(ContainerBuilder $container): void
+    {
+        // Iterate through all service definitions
+        foreach ($container->getDefinitions() as $definition) {
+            $class = $definition->getClass();
+
+            // Skip if class is not set or is a parameter
+            if (!$class || str_starts_with($class, '%')) {
+                continue;
+            }
```

Comment:
```suggestion
            if (!$class) {
                continue;
            }
```
not entirely sure if this condition would get some false positive, seems that would be just fine to `resolveValue()` regardless as the try-catch below

Reply:
@kingjia90 Good catch, thanks! 😊 I will fix and optimize this.